src/main/resources/cpp-modern-json/model-header.mustache

```diff
@@ -1,129 +1,12 @@
 {{#models}}{{#model}}#ifndef {{modelHeaderGuardPrefix}}_{{classname}}_H_
 #define {{modelHeaderGuardPrefix}}_{{classname}}_H_
 
-#include <nlohmann/json.hpp>
-
-{{#hasOptional}}#include <optional>
-{{/hasOptional}}
-
-{{#imports}}{{{this}}}
-{{/imports}}
-
-{{! ---------------------------------------------------------------------- }}
-{{! the model struct                                                       }}
-{{! ---------------------------------------------------------------------- }}
-
-{{#modelNamespaceDeclarations}}
-namespace {{this}} {
-{{/modelNamespaceDeclarations}}
-
-{{#description}}/**
- * \brief {{description}}
-{{#example}} *
- * Example: {{example}}
-{{/example}}
- */
-{{/description}}
-struct {{classname}} {
-
-    static void to_json(::nlohmann::json& j, const {{classname}}& obj);
-    static void from_json(const ::nlohmann::json& j, {{classname}}& obj);
-
-{{#hasEnums}}    // enums
-{{#vars}}{{#isEnum}}{{#description}}    /**
-     * \brief {{description}}
-     */
-{{/description}}
-    enum class {{enumName}} {
-{{#allowableValues}}{{#enumVars}}        {{name}},
-{{/enumVars}}{{/allowableValues}}    };
-
-    static std::string to_string({{enumName}} value);
-    static void to_json(::nlohmann::json& j, const {{enumName}}& value);
-    static void from_json(const ::nlohmann::json& j, {{enumName}}& value);
-
-{{/isEnum}}
-{{/vars}}
-{{/hasEnums}}
-
-{{#hasRequired}}    // required fields
-{{#requiredVars}}    {{{datatypeWithEnum}}} {{baseName}};
-{{/requiredVars}}
-{{/hasRequired}}
-
-{{#hasOptional}}    // optional fields
-{{#optionalVars}}    std::optional<{{{datatypeWithEnum}}}> {{baseName}};
-{{/optionalVars}}
-{{/hasOptional}}
-};
-
-{{! ---------------------------------------------------------------------- }}
-{{! to_string methods for enum fields                                      }}
-{{! ---------------------------------------------------------------------- }}
-
-{{#hasEnums}}
-{{#vars}}
 {{#isEnum}}
-inline std::string to_string({{classname}}::{{enumName}} value) {
-    return {{classname}}::to_string(value);
-}
-
+    {{>enumClass}}
```

Comment:
I like this approach very much. Because of this, it will be easier to maintain in the future. Thx!